src/components/search/SearchService.js

```diff
@@ -100,31 +110,37 @@ goog.provide('ga_search_service');
           position =
             [left > right ? left : right,
               right < left ? right : left];
-          if (ol.extent.containsCoordinate(
-              extent, position)) {
-            valid = true;
+          if (ol.extent.containsCoordinate(extent, position)) {
+            defer.resolve(roundCoordinates(position));
           } else {
-            position = ol.proj.transform(position,
-              'EPSG:2056', 'EPSG:21781');
-            if (ol.extent.containsCoordinate(
-                extent, position)) {
-              valid = true;
-            } else {
-              position =
-                [left < right ? left : right,
-                  right > left ? right : left];
-              position = ol.proj.transform(position,
-                'EPSG:4326', 'EPSG:21781');
-              if (ol.extent.containsCoordinate(
-                extent, position)) {
-                valid = true;
+            gaReframe.get95To03(position).then(function(response) {
+              position = response.data.coordinates;
+              if (ol.extent.containsCoordinate(extent, position)) {
+                defer.resolve(roundCoordinates(position));
+              } else {
+                position =
+                  [left < right ? left : right,
+                    right > left ? right : left];
+                position = ol.proj.transform(position,
+                  'EPSG:4326', 'EPSG:21781');
+                if (ol.extent.containsCoordinate(extent, position)) {
+                  defer.resolve(roundCoordinates(position));
+                } else {
+                  defer.resolve(undefined);
+                }
               }
-            }
+            }, function() {
+              // On error use proj4js
+              position = ol.proj.transform(position, 'EPSG:2056', 'EPSG:21781');
```

Comment:
Could we move the proj4 transformation in case of error in the reframe service ?
with that we could get rid of the else line 120 .